### DISCUSSION TITLE
Apply instances to all frames?

### DISCUSSION THREAD

**Discussion by ginapastina** [talmolab/sleap#1248] [Help!]
Hi, 
I am new to SLEAP and these types of programs in general, but I am working on a large project for my lab. I was wondering if, after you run inference, is there a way to apply all of the inferences at once? I keep clicking "add instances from all predictions to current frame" but I was hoping there was a quicker and easier way to just put all of the predicted instances to the entire video at once. Thank you so much!

**Comment by adam-matic** [comment]
Not sure if I understand - I just double-click each correctly predicted instance in the frame and leave the incorrectly labeled instance in yellow. If all the instances are correct, I don't add that frame to the training set, I'm thinking the program already has enough examples of those frames.  Also, I only have 5-6 instances maximum per frame.

**Reply by ginapastina** [reply]
Hi! Thank you for replying.

I meant just a general way to add all suggested instances to the entire video not per frame, like how you are able to delete all user made instances, I am trying to see if there is something for adding the instances that were predicted by the program that applies to every frame. I am only working with 3 instances, so that is good. When you say you don't add the frame to the training set when all the instances are correct? Sorry I am super super new to stuff like this so I am always looking to learn better ways to do all this. Thank you!!

**Reply by roomrys** [reply]
Hi @adam-matic,

That sounds like a good workflow! Just wanted to pipe-in and make sure that you are either fully annotating a frame or not annotating the frame at all. Meaning, that if you correct one predicted instance in a frame, you should be correcting/converting all the predictions to user-labeled instances. Otherwise you are essentially teaching SLEAP to ignore some instances.

Thanks for contributing to the community!
Liezl

**Reply by adam-matic** [reply]
Hi, Liezl! Yeah, sure, I convert all the correctly predicted instances to labels, so the frame is fully annotated.

**Comment by roomrys** [comment]
Hi @ginapastina,

We do not have a feature for converting all predicted instances to user instances. We avoided adding this feature because users typically need to verify that the predictions are correct and good enough to be used in training (or correct them before retraining). Otherwise, you may get a poor performing model because there are poorly labeled frames in the project.

Thanks,
Liezl

**Reply by ginapastina** [reply]
Hi Liezl,

Thank you for clarifying! I appreciate you taking time to respond. 

Gina
